packages/room-server/src/fusion/services/fusion.api.service.ts

```diff
@@ -502,21 +503,15 @@ export class FusionApiService {
     if (result.result !== ExecuteResult.Success) {
       throw ApiException.tipError(ApiTipConstant.api_update_error);
     }
-
-    const newDatasheet = await this.databusService.getDatasheet(dstId, {
-      loadOptions: {
-        auth,
-        recordIds,
-        linkedRecordMap,
-      },
-    });
-    if (newDatasheet === null) {
-      throw ApiException.tipError(ApiTipConstant.api_datasheet_not_exist);
-    }
+    // success doesn't mean that all records are updated successfully, could be partial success
+    // such as the field type is changed while updating, the value may be invalid
+    // so we need to reload the record map to get the correct value
+    const recordMap = await this.fusionApiRecordService.getBasicRecordsByRecordIds(dstId, recordIds);
```

Comment:
This approach bypasses DataBus. It is better to handle this case in the data loader and pass an extra option (e.g. `loadOptions: { loadRecordOnly: true }`) to `database.getDatasheet()`.
Besides, if a record contains `Lookup` fields, foreign datasheet map is required to calculate their cell values. If the record contains `AutoNum`, `LastModifiedTime`, `LastModifiedBy`, `CreatedTime`, or `CreatedBy` fields, `recordMeta` is required to get their cell value.